api/src/main/java/org/openmrs/module/appointments/service/impl/RecurringAppointmentServiceImpl.java

```diff
@@ -120,12 +123,73 @@ public void changeStatus(Appointment appointment, String status, Date onDate, St
                 .collect(Collectors.toList());
     }
 
-    private Appointment updateMetadata(Appointment pendingAppointment, Appointment appointment) {
+    private void updateMetadata(Appointment pendingAppointment, Appointment appointment) {
         Date startTime = getStartTime(pendingAppointment, appointment);
         Date endTime = getEndTime(pendingAppointment, appointment);
         pendingAppointment.setStartDateTime(startTime);
         pendingAppointment.setEndDateTime(endTime);
-        return pendingAppointment;
+        pendingAppointment.setService(appointment.getService());
+        pendingAppointment.setServiceType(appointment.getServiceType());
+        pendingAppointment.setLocation(appointment.getLocation());
+        pendingAppointment.setComments(appointment.getComments());
+        if (appointment.getProviders() != null) {
+            mapProvidersForAppointment(pendingAppointment,
+                    getDeepCloneOfProviders(new ArrayList<>(appointment.getProviders())));
+        }
+    }
+
+    private List<AppointmentProvider> getDeepCloneOfProviders(List<AppointmentProvider> appointmentProviders) {
+        return appointmentProviders
+                .stream()
+                .map(AppointmentProvider::new)
+                .collect(Collectors.toList());
+    }
+
+    private void mapProvidersForAppointment(Appointment appointment, List<AppointmentProvider> newProviders) {
+        Set<AppointmentProvider> existingProviders = appointment.getProviders();
+        setRemovedProvidersToCancel(newProviders, existingProviders);
+        createNewAppointmentProviders(appointment, newProviders);
+    }
+
+    private void createNewAppointmentProviders(Appointment appointment, List<AppointmentProvider> newProviders) {
+        if (newProviders != null && !newProviders.isEmpty()) {
```

Comment:
Use StringUtils.isEmpty() utility from apache commons. Update in other places also.

Reply:
used Collectionutils.isempty where ever we are verifying if it's null/empty